azure/Kqlmagic/draft_client.py

```diff
@@ -41,14 +41,45 @@ class DraftClient(object):
     _API_VERSION = "v1"
     _GET_SCHEMA_QUERY = ".show schema"
 
-    def __init__(self, conn_kv: dict, domain: str, data_source: str):
+    _CLOUD_AAD_URLS={
+    "public": "https://api.{0}.io",
+    "mooncake":"https://api.{0}.azure.cn",
+    "fairfax":"https://api.{0}.us",
+    "blackforest":"https://api.{0}.de",
+}
+
+
+
+    def __init__(self, conn_kv: dict, domain: str, data_source: str, **options):
+
+        cloud = options.get("cloud") or "public"
         self._domain = domain
-        self._data_source = data_source
+        logger().debug("draft_client.py :: __init__ :  self._domain: {0}".format(self._domain))
+
+        if data_source.find("loganalytics") >= 0:
+            self._data_source = self._CLOUD_AAD_URLS.get(cloud).format("loganalytics")
+            logger().debug("draft_client.py :: __init__ :  self._data_source from _CLOUD_AAD_URLS: {0}".format(self._data_source))
+
+        elif data_source.find("applicationinsights") >= 0:
+            self._data_source = self._CLOUD_AAD_URLS.get(cloud).format("applicationinsights")  
+            logger().debug("draft_client.py :: __init__ :  self._data_source from _CLOUD_AAD_URLS: {0}".format(self._data_source))
+
+
+        logger().debug("draft_client.py :: __init__ :  conn_kv[\"datasourceurl\"]: {0}".format(conn_kv.get("datasourceurl")))
+
+        if conn_kv.get("datasourceurl"):
```

Comment:
the code above us misleading, there should be at the begining if conn_kv.get("datasourceurl")... elsif..